### PR TITLE
Doc: rewrite and expand bookmarking section

### PR DESCRIPTION
**Pull request by Minoru**
@christosmarg, this PR is inspired by our chat on IRC yesterday; please take a look and tell me if the new version is any clearer than the old one.

Requesting a review from @der-lyse since I run all the docs by him, but everyone else is welcome to review too! I'll merge in three days if no issues are raised.

### CODE REVIEW

File: doc/newsboat.asciidoc [hunk 1]
```diff
@@ -740,19 +740,47 @@ include::chapter-snownews.asciidoc[]
...
-This error message is then presented to the user from within Newsboat.
-
-Newsboat comes with an example plugin, which implements a simple tab-separated bookmark file. This
-example can be found in the _doc/examples_ subdirectory.
+This feature lets you save links from the article list, the article view, and
+the URL view. The actual bookmarking is performed by a program that you specify
+via the <<bookmark-cmd,`bookmark-cmd`>> setting; Newsboat merely supplies the
+data.
+
+To bookmark the currently selected item, press `^B` (invoking
```
Comment (der-lyse) [inline doc/newsboat.asciidoc:748]: I reckon there should be an article:

```suggestion
To bookmark the currently selected item, press `^B` (invoking the
```

File: doc/newsboat.asciidoc [hunk 2]
```diff
@@ -740,19 +740,47 @@ include::chapter-snownews.asciidoc[]
...
+4. (since Newsboat 2.10) the title of the feed you're currently in (preset as
+   you'd expect).
+
+(If you find that the above preset values always work for you, enable
+<<bookmark-autopilot,`bookmark-autopilot`>> to avoid being asked anything.)
+
+After that, the program configured via `bookmark-cmd` is executed. It is given
+four arguments, the same ones and in the same order as described above. The
+program then does the actual bookmark saving, e.g. writing the bookmark into an
+external file, or storing it to the del.icio.us account.
```
Comment (der-lyse) [inline doc/newsboat.asciidoc:764]: I think the indefinite article from the original text reads a wee bit nicer.

```suggestion
external file, or storing it to a del.icio.us account.
```

**Review by der-lyse — APPROVED** [review]
This is brilliant!

**Review by der-lyse — APPROVED** [review]
This is brilliant!

**Comment by christosmarg** [comment]
All good :-)

**Comment by Minoru** [comment]
Thanks for the review! I applied both your suggestions, @der-lyse, and will squash everything when merging.

I'll leave this open for a while to give others a chance to leave comments of their own.

**Review by dennisschagt — APPROVED** [review]
LGTM

**Comment by Minoru** [comment]
With three reviewers this should be fine :) Thank you all. Merging!

**Pull request merged**